### PR TITLE
Add missing delete_on_termination passthrough. Adapt docs.

### PR DESCRIPTION
**Pull request by nasenbaer13**
### What does this PR do?

Passthrough delete_on_termination. This was documented but not implemented.

### What issues does this PR fix or reference?

no reported one, see above.

### Tests written?

No

### CODE REVIEW

**Comment by nasenbaer13** [comment]
Please also merge in 2017.7 if possible

**Comment by cachedout** [comment]
@saltstack/team-boto can y'all please come past here and help with a review, if you have a free moment?

**Review by tkwilliams — APPROVED** [review]
Good catch :)

**Review by tkwilliams — APPROVED** [review]
Looks good, thanks!

File: salt/states/boto_lc.py [hunk 1]
```diff
@@ -122,6 +122,7 @@ def present(
         kernel_id=None,
         ramdisk_id=None,
         block_device_mappings=None,
+        delete_on_termination=True,
```
Comment (rallytime) [inline salt/states/boto_lc.py]: @nasenbaer13 My concern here is that this results in a behavior change. I know it was documented, but it wasn't implemented as you said. So when folks upgrade to Oxygen, these will start deleting when they didn't before. Perhaps that is still desired, but I wanted to at least bring it up.

What do you think @ryan-lane?

Comment (nasenbaer13) [reply salt/states/boto_lc.py]: So @tkwilliams - I understand you correctly: changing the `delete_on_termination` default to `None` and pass it through should do the trick? At least it shouldn't change the previous behaviour right? But at least it is possible to change it.

Comment (nasenbaer13) [reply salt/states/boto_lc.py]: PR updated, @tkwilliams does it look better now?

**Comment by nasenbaer13** [comment]
@tkwilliams, updated. Now using None as default and docs are adapted.

**Review by rallytime — APPROVED** [review]

**Comment by rallytime** [comment]
re-run py3

File: salt/states/boto_lc.py [hunk 2]
```diff
@@ -185,9 +186,10 @@ def present(
             Indicates what volume type to use. Valid values are standard, io1, gp2.
             Default is standard.
 
-        delete_on_termination
-            Indicates whether to delete the volume on instance termination (true) or
-            not (false).
+    delete_on_termination
+        Indicates whether to delete the volume on instance termination (true) or
+        not (false). Default is "None" which corresponds to not specified.
+        Amazon has different defaults for root and additional volumes.
```
Comment (tkwilliams) [inline salt/states/boto_lc.py]: To be pedantic for the sake of least surprise for end users, I'd recommend something like
```
Whether the volume should be explicitly marked for deletion when its instance is
terminated (True), or left around (False).  If not provided, or None is explicitly passed,
the default AWS behaviour is used, which is True for ROOT volumes of instances, and
False for all others.
```

Or something to that effect... :)

**Review by tkwilliams — CHANGES_REQUESTED** [review]

File: salt/states/boto_lc.py (hunk 2 above)
Comment (nasenbaer13) [reply salt/states/boto_lc.py]: Sounds good, changed in last commit.

**Comment by rallytime** [comment]
Thank you for this @nasenbaer13 and thanks for the review help @tkwilliams. I think this is ready to go in now. 🎉

**Comment by nasenbaer13** [comment]
@rallytime shall I open another pr rebased on the 2017.7.  branch?

**Comment by rallytime** [comment]
@nasenbaer13 That would be great!

**Comment by nasenbaer13** [comment]
opened #44054, @rallytime you can close this one I guess.

**Pull request merged**